napari/_vispy/vispy_scale_bar_visual.py

```diff
@@ -117,57 +145,42 @@ def _on_data_change(self, event):
         self.text_node.color = color
 
     def _on_visible_change(self, event):
-        """Change visibiliy of scale bar."""
+        """Change visibility of scale bar."""
         self.node.visible = self._viewer.scale_bar.visible
         self.text_node.visible = self._viewer.scale_bar.visible
         self._on_zoom_change(None)
 
-    def _on_position_change(self, event):
+    def _on_text_change(self, event):
+        """Update text information"""
+        self.text_node.font_size = self._viewer.scale_bar.font_size
+
+    def _on_position_change(self, _evt=None):
         """Change position of scale bar."""
-        if self._viewer.scale_bar.position == Position.TOP_LEFT:
+        position = self._viewer.scale_bar.position
+        x_bar_offset = 10
+        canvas_size = list(self.node.canvas.size)
+
+        if position == Position.TOP_LEFT:
             sign = 1
-            self.node.transform.translate = [66, 14, 0, 0]
-            self.text_node.transform.translate = [33, 16, 0, 0]
-        elif self._viewer.scale_bar.position == Position.TOP_RIGHT:
+            bar_transform = [x_bar_offset, 10, 0, 0]
+        elif position == Position.TOP_RIGHT:
             sign = -1
-            canvas_size = list(self.node.canvas.size)
-            self.node.transform.translate = [canvas_size[0] - 66, 14, 0, 0]
-            self.text_node.transform.translate = [
-                canvas_size[0] - 33,
-                16,
-                0,
-                0,
-            ]
-        elif self._viewer.scale_bar.position == Position.BOTTOM_RIGHT:
+            bar_transform = [canvas_size[0] - x_bar_offset, 10, 0, 0]
+        elif position == Position.BOTTOM_RIGHT:
             sign = -1
-            canvas_size = list(self.node.canvas.size)
-            self.node.transform.translate = [
-                canvas_size[0] - 66,
-                canvas_size[1] - 16,
+            bar_transform = [
+                canvas_size[0] - x_bar_offset,
+                canvas_size[1] - 30,
                 0,
                 0,
             ]
-            self.text_node.transform.translate = [
-                canvas_size[0] - 33,
-                canvas_size[1] - 14,
-                0,
-                0,
-            ]
-        elif self._viewer.scale_bar.position == Position.BOTTOM_LEFT:
+        elif position == Position.BOTTOM_LEFT:
             sign = 1
-            canvas_size = list(self.node.canvas.size)
-            self.node.transform.translate = [66, canvas_size[1] - 16, 0, 0]
-            self.text_node.transform.translate = [
-                33,
-                canvas_size[1] - 14,
-                0,
-                0,
-            ]
+            bar_transform = [x_bar_offset, canvas_size[1] - 30, 0, 0]
```

Comment:
What does this `30` refer to 🤔 ?

Reply:
Is it `y_bar_offset`? Maybe we make that a variable like `x_bar_offset`